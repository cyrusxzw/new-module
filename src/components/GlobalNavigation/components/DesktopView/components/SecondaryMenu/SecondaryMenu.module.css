@import 'styles/index';

.base {
  pointer-events: none;
  transform: translateX(5px);
}

.list {
  display: flex;
  padding-left: 0;
  margin-top: 0;
  margin-bottom: 0;
  list-style-type: none;
  pointer-events: auto;
}
