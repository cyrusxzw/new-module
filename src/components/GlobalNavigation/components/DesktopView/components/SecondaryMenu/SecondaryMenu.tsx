import React from 'react';
import { useGlobalNavigationContext } from '~/components/GlobalNavigation/GlobalNavigation.context';
import { MenuItem } from '../MenuItem';
import type { SecondaryMenuType } from './SecondaryMenu.types';
import styles from './SecondaryMenu.module.css';

const SecondaryMenu: SecondaryMenuType = () => {
  const {
    actions: { account, cart },
  } = useGlobalNavigationContext();

  return (
    <nav aria-label="supplementary" className={styles.base} role="navigation">
      <ul className={styles.list}>
        <MenuItem
          dataTestRef={account.dataTestRef}
          id={account.id}
          label={account.label}
          onClick={account.type === 'trigger' ? account.onClick : null}
          title={account.title}
          type={account.type}
          url={account.type === 'link' ? account.url : null}
        />

        <MenuItem
          dataTestRef={cart.dataTestRef ?? 'NAV_CART'}
          id={cart.id}
          label={cart.label}
          onClick={cart.onClick}
          title={cart.title}
          type={cart.type}
        />
      </ul>
    </nav>
  );
};

export { SecondaryMenu };
