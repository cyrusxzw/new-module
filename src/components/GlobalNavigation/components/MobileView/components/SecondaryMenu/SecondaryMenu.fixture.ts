import type { SecondaryMenuProps } from './SecondaryMenu.types';

const SecondaryMenuFixture: SecondaryMenuProps = {
  items: [
    {
      id: 'link-one',
      label: 'Link One',
      title: 'Link One',
      type: 'link' as const,
      url: '#link-one',
    },
    {
      id: 'link-two',
      label: 'Link Two',
      title: 'Link Two',
      type: 'link' as const,
      url: '#link-two',
    },
  ],
};

export { SecondaryMenuFixture };
