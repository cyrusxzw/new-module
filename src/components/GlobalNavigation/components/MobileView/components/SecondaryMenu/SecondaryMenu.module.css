@import 'styles/index';


.list {
  padding-right: 15px;
  padding-left: 15px;
  margin-top: 30px;
  margin-bottom: 30px;
  list-style: none;
}

.item {
  display: inline-block;
  width: 50%;
}

.element {
  text-align: left;
  padding-top: 10px;
  padding-right: 5px;
  padding-bottom: 10px;
  padding-left: 5px;
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(14px);
  opacity: 1;
  outline-offset: 0;
  transition: opacity 200ms var(--easing-ease-out-cubic);
  transition: opacity 200ms var(--easing-ease-out-cubic),
    outline-offset 200ms var(--easing-ease-out-cubic),
    color 200ms var(--easing-ease-out-cubic);

  &.hidden {
    opacity: 0;
    pointer-events: none;
  }

  &:hover {
    .ornamentalHover::after {
      width: 100%;
    }
  }

  .dark & {
    color: var(--color-dark-label);
  }

  .light & {
    color: var(--color-light-label);
  }

  .dark &:hover {
    color: var(--color-black);
  }

  .light &:hover {
    color: var(--color-white);
  }

  .dark &:focus-visible {
    outline: 1px dashed var(--color-dark-decoration);
    outline-offset: 2px;
  }

  .light &:focus-visible {
    outline: 1px dashed var(--color-dark-decoration);
    outline-offset: 2px;
  }
}

.stores {
  z-index: calc(var(--z-index-fixed-menu) + 1);
  height: 100%;

  .dark & {
    border-top: 1px solid var(--color-grey-60);
    background-color: var(--color-primary);
  }

  .light & {
    border-top: 1px solid var(--color-white);
    background-color: var(--color-secondary);
  }
}
