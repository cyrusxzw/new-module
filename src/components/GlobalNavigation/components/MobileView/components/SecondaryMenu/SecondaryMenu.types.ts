import type { ComponentWithoutChildren } from '~/types';
import type {
  Trigger,
  Link,
  Read,
  MenuItemTrackingWithMobileAction,
} from '~/components/GlobalNavigation/GlobalNavigation.types';

type SecondaryMenuItem = Trigger | Link | Read;

type SecondaryMenuProps = {
  isVisible?: boolean;
  items?: SecondaryMenuItem[];
};

type SecondaryMenuType = ComponentWithoutChildren<SecondaryMenuProps>;

type MenuItemTrackingWithMobileActionType = MenuItemTrackingWithMobileAction;

export type {
  MenuItemTrackingWithMobileActionType,
  SecondaryMenuType,
  SecondaryMenuProps,
  SecondaryMenuItem,
};
