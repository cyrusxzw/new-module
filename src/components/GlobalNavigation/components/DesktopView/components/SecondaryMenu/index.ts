import { SecondaryMenu } from './SecondaryMenu';

export { SecondaryMenu };
