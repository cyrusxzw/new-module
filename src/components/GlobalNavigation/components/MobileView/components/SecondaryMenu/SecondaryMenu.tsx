import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import {
  useGlobalNavigationStateContext,
  useGlobalNavigationContext,
} from '~/components/GlobalNavigation/GlobalNavigation.context';
import { Button } from '~/components/Button';
import { Hyperlink } from '~/components/Hyperlink';
import { Transition } from '~/components/Transition';
import { Collection } from '../Collection';
import type {
  SecondaryMenuType,
  SecondaryMenuItem,
} from './SecondaryMenu.types';
import compositionStyles from '../../MobileView.module.css';
import styles from './SecondaryMenu.module.css';

const SecondaryMenu: SecondaryMenuType = ({ items }) => {
  const {
    activeCollectionId,
    isOpen,
    setActiveCollectionId,
  } = useGlobalNavigationStateContext();

  const {
    actions: { stores },
  } = useGlobalNavigationContext();

  const currentTheme = useThemeContext(null, 'dark');

  if (!items?.length) return null;

  const isVisible = isOpen && activeCollectionId === 'top';
  const handleSetActiveCollectionId = (id: string) => setActiveCollectionId(id);
  const classSet = cx(styles.base, styles[currentTheme]);

  const elementClassName = cx(
    styles.element,
    { [styles.hidden]: activeCollectionId !== 'top' },
    compositionStyles.ornamentalWrapper,
  );

  const handleOnStoresClick = () => {
    stores?.onClick();
    setActiveCollectionId(stores.id);
  };

  const getElement = (props: SecondaryMenuItem) => {
    if (!props) return null;

    const handleOnClick = () => {
      if (props.type === 'read-collection') {
        handleSetActiveCollectionId(props.id);
      } else if (props.id === stores.id) {
        handleOnStoresClick();
      } else if (props.type === 'trigger') {
        props.onClick();
      }
    };

    return props.type === 'link' ? (
      <Hyperlink
        className={elementClassName}
        dataTestRef={props.dataTestRef}
        tabIndex={!isVisible ? -1 : null}
        title={props.title}
        url={props.url}
      >
        <span className={compositionStyles.ornamentalHover}>{props.label}</span>
      </Hyperlink>
    ) : (
      <>
        <Button
          className={elementClassName}
          dataTestRef={props.dataTestRef}
          isInline={true}
          onClick={handleOnClick}
          tabIndex={!isVisible ? -1 : null}
          title={props.title}
        >
          <span className={compositionStyles.ornamentalHover}>
            {props.label}
          </span>
        </Button>

        {props.type === 'read-collection' && (
          <Collection
            {...props}
            isVisible={activeCollectionId === 'read-collection'}
            shouldHideTopCollection={true}
          />
        )}

        {props.id === stores.id && (
          <Transition
            isActive={activeCollectionId === stores.id}
            shouldMountOnEnter={true}
            shouldUnmountOnExit={true}
            type="slideLeft"
          >
            <div className={cx(compositionStyles.slide, styles.stores)}>
              <stores.component />
            </div>
          </Transition>
        )}
      </>
    );
  };

  return (
    <nav
      aria-hidden={!isVisible}
      aria-label="supplementary"
      className={classSet}
      role="navigation"
    >
      <ul className={styles.list}>
        {items.filter(Boolean).map((props: SecondaryMenuItem) => {
          return (
            <li className={styles.item} key={props.label}>
              {getElement(props)}
            </li>
          );
        })}
      </ul>
    </nav>
  );
};

export { SecondaryMenu };
