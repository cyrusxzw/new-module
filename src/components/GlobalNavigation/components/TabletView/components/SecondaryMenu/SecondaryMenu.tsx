import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Button } from '~/components/Button';
import { Hyperlink } from '~/components/Hyperlink';
import { useGlobalNavigationContext } from '~/components/GlobalNavigation/GlobalNavigation.context';
import { MenuItem } from '../MenuItem';
import type { SecondaryMenuType } from './SecondaryMenu.types';
import compositionStyles from '../../TabletView.module.css';
import styles from './SecondaryMenu.module.css';

const SecondaryMenu: SecondaryMenuType = () => {
  const {
    actions: { account, cart },
  } = useGlobalNavigationContext();

  const currentTheme = useThemeContext(undefined, 'dark');

  const { dataTestRef, id, title } = account;

  const sharedAccountProps = {
    className: cx(styles.element, compositionStyles.ornamentalWrapper),
    dataTestRef,
    id,
    title,
  };

  const classSet = cx(compositionStyles.menuList, styles[currentTheme]);

  return (
    <nav aria-label="supplementary" role="navigation">
      <ul className={classSet}>
        <MenuItem>
          {account.type === 'link' ? (
            <Hyperlink {...sharedAccountProps} url={account.url}>
              <span className={compositionStyles.ornamentalHover}>
                {account.label}
              </span>
            </Hyperlink>
          ) : (
            <Button
              {...sharedAccountProps}
              isInline={true}
              onClick={account.onClick}
            >
              <span className={compositionStyles.ornamentalHover}>
                {account.label}
              </span>
            </Button>
          )}
        </MenuItem>
        <MenuItem>
          <Button
            className={cx(styles.element, compositionStyles.ornamentalWrapper)}
            dataTestRef={cart.dataTestRef ?? 'NAV_CART'}
            id={cart.id}
            isInline={true}
            onClick={cart.onClick}
            title={cart.title}
          >
            <span className={compositionStyles.ornamentalHover}>
              {cart.label}
            </span>
          </Button>
        </MenuItem>
      </ul>
    </nav>
  );
};

export { SecondaryMenu };
