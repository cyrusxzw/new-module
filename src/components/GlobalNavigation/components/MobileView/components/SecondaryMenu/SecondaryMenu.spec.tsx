import React from 'react';
import { render } from '@testing-library/react';
import { SecondaryMenu } from './SecondaryMenu';
import { SecondaryMenuFixture } from './SecondaryMenu.fixture';

import {
  GlobalNavigationStateContextProvider,
  useGlobalNavigationStateContext,
  useGlobalNavigationContext,
} from '~/components/GlobalNavigation/GlobalNavigation.context';

jest.mock('~/components/GlobalNavigation/GlobalNavigation.context');

describe('<GlobalNavigation.MobileView.SecondaryMenu />', () => {
  const mockActions = {
    stores: { component: 'stores', id: 'stores' },
    search: { component: 'search', id: 'search' },
    cart: 'cart',
    menu: 'menu',
  };

  const testGlobalNavigationContext = {
    actions: mockActions,
    onOpen: jest.fn(),
  };

  const testGlobalNavigationStateContext = {
    activeCollectionId: 'top',
    setActiveCollectionId: jest.fn(),
    isOpen: false,
    setIsOpen: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
    (useGlobalNavigationStateContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationStateContext,
    });
    (useGlobalNavigationContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationContext,
    });
  });

  const TestBed = ({ items }: { items?: any }) => {
    return (
      <GlobalNavigationStateContextProvider>
        <SecondaryMenu {...SecondaryMenuFixture} items={items} />
      </GlobalNavigationStateContextProvider>
    );
  };

  it('should render base component correctly', () => {
    const { container } = render(
      <TestBed items={SecondaryMenuFixture.items} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should return null with no items passed', () => {
    const { container } = render(<TestBed />);

    expect(container).toBeEmptyDOMElement();
  });

  // TODO: Identify way to capture uncovered branches
});
