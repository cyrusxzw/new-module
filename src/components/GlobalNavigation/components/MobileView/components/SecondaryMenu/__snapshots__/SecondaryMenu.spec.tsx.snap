// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalNavigation.MobileView.SecondaryMenu /> should render base component correctly 1`] = `
<div>
  <nav
    aria-hidden="true"
    aria-label="supplementary"
    class="base dark"
    role="navigation"
  >
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <a
          class="base left dark element ornamentalWrapper"
          href="#link-one"
          tabindex="-1"
          target="_self"
          title="Link One"
        >
          <span
            class="ornamentalHover"
          >
            Link One
          </span>
        </a>
      </li>
      <li
        class="item"
      >
        <a
          class="base left dark element ornamentalWrapper"
          href="#link-two"
          tabindex="-1"
          target="_self"
          title="Link Two"
        >
          <span
            class="ornamentalHover"
          >
            Link Two
          </span>
        </a>
      </li>
      <li
        class="item"
      >
        <button
          class="base dark element ornamentalWrapper"
          tabindex="-1"
          title="Collection One"
          type="button"
        >
          <span
            class="ornamentalHover"
          >
            Collection One
          </span>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
